u8timeseries/models/autoregressive_model.py

```diff
@@ -31,7 +33,7 @@ def predict(self, n: int) -> TimeSeries:
         """
         :return: A TimeSeries containing the `n` next points, starting after the end of the training time series.
         """
-        assert self._fit_called, 'fit() must be called before predict()'
+        assert_log(self._fit_called, 'fit() must be called before predict()', logger)
```

Comment:
Here `ValueError` is probably not a good fit, maybe there is a better exception type? (Other than AssertError or ValueError)

Reply:
used generic Exception, done